card/mod/history/lib/card/act.rb

```diff
@@ -73,10 +73,13 @@ def actor
     # the act's primary card
     # @return [Card]
     def card
-      res = Card.fetch card_id, look_in_trash: true, skip_modules: true
-      return res unless res&.type_id&.in?([FileID, ImageID])
+      Card.fetch card_id, look_in_trash: true #, skip_modules: true
 
-      res.include_set_modules
+      # FIXME: if the following is necessary, we need to document why.
+      # generally it's a very bad idea to have type-specific code here.
```

Comment:
Why is it bad? What's the alternative? This is an AR class not a card set and `res` doesn't include set modules.

If you check the git logs you see that this was introduced to fix broken files and images in the history. Since `skip_modules: true` was probably there only for performance reasons it should be safe to remove it.

Reply:
it's bad because as soon as anybody creates a new type/Image class, it will break.  Generally speaking, if we hardcode type handling, we handcuff monkeys.

Reply:
Agree re performance.  I suspect (but have not confirmed) that this was the ultimate source of the reversion error.  (Though the error itself was not as big a deal as the over-reversion.